powershell/VstsTaskSdk/InputFunctions.ps1

```diff
@@ -158,60 +160,121 @@ function Get-EnvVariable {
+    [CmdletBinding(DefaultParameterSetName = 'Require')]
+    param(
+        [string]$Value,
+        [Parameter(Mandatory = $true)]
+        [string]$Name,
+        [Parameter(Mandatory = $true)]
+        [string]$Path,
+        [Parameter(ParameterSetName = 'Require')]
+        [switch]$Require,
+        [Parameter(ParameterSetName = 'Default')]
+        [object]$Default,
+        [switch]$AsBool,
+        [switch]$AsInt)
+
+    $result = $Value
+    if ($result) {
+        if ($Path -like 'Env:ENDPOINT_AUTH_*') {
+            Write-Verbose "$($Path): '********'"
+        } else {
+            Write-Verbose "$($Path): '$result'"
+        }
     } else {
-        Write-Verbose "$path (empty)"
+        Write-Verbose "$Path (empty)"
 
         # Write error if required.
         if ($Require) {
-            # TODO: This is technically wrong for loc. It would be best to grab the
-            # localized description from the json. However, this might be OK since
-            # this type of validation is typically performed up-front by the UI. So
-            # the likely scenario for encountering this error would be if someone
-            # (a developer) uploaded a definition themselves. At that point it should
-            # be upon them to validate what they are doing. And since we don't localize
-            # for developers, this may not matter after all.
             Write-Error "$(Get-LocString -Key PSLIB_Required0 $Name)"
             return
         }
 
         # Fallback to the default if provided.
         if ($PSCmdlet.ParameterSetName -eq 'Default') {
-            $value = $Default
-            Write-Verbose " Defaulted to: '$value'"
+            $result = $Default
+            Write-Verbose " Defaulted to: '$result'"
         } else {
-            $value = ''
+            $result = ''
         }
     }
 
     # Convert to bool if specified.
     if ($AsBool) {
-        if ($value -isnot [bool]) {
-            $value = "$value" -in '1', 'true'
-            Write-Verbose " Converted to bool: $value"
+        if ($result -isnot [bool]) {
+            $result = "$result" -in '1', 'true'
+            Write-Verbose " Converted to bool: $result"
         }
 
-        return $value
+        return $result
     }
 
     # Convert to int if specified.
     if ($AsInt) {
-        if ($value -isnot [int]) {
+        if ($result -isnot [int]) {
             try {
-                $value = [int]"$value"
+                $result = [int]"$result"
             } catch {
-                $value = 0
+                $result = 0
             }
 
-            Write-Verbose " Converted to int: $value"
+            Write-Verbose " Converted to int: $result"
         }
 
-        return $value
+        return $result
     }
 
-    return $value
+    return $result
+}
+
+function Initialize-SecureInputs {
+    # Store endpoints/inputs in a secure fashion.
+    foreach ($variable in (Get-ChildItem -Path Env:ENDPOINT_*, Env:INPUT_*)) {
```

Comment:
is this means we will put all inputs into security string, even the input is not a secret?